File: SportsStoreVb/SportsStoreVb.WebUI/Controllers/ProductController.vb
Imports System.Web.Mvc
Imports System.Linq
Imports SportsStoreVb.Domain

Namespace Controllers
    Public Class ProductController
        Inherits Controller

        ReadOnly repository As IProductRepository
        Public PageSize As Integer = 4

        Public Sub New (productRepository As IProductRepository)
            Me.repository = productRepository
        End Sub

        ' GET: Product
        'Function Index() As ActionResult
        '    Return View()
        'End Function

        ' Public Function List() As ViewResult
        Public Function List(Optional page As Integer = 1) As ViewResult
            ' Return View(repository.Products)
            Return View(repository.Products _
                        .OrderBy(Function(p) p.ProductID) _
                        .Skip((page - 1) * PageSize) _
                        .Take(PageSize)
                        )
        End Function
    End Class
End Namespace

File: SportsStoreCs/SportsStoreCs.WebUI/Controllers/ProductController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace SportsStoreCs.WebUI.Controllers
{
    using Domain.Abstract;

    public class ProductController : Controller
    {
        private readonly IProductRepository repository;
        public int PageSize = 4;

        public ProductController(IProductRepository productRepository)
        {
            this.repository = productRepository;
        }

        // GET: Product
        //public ActionResult Index()
        //{
        //    return View();
        //}

        // public ViewResult List()
        public ViewResult List(int page = 1)
        {
            // return View(repository.Products);
            return View(repository.Products
                .OrderBy(p => p.ProductID)
                .Skip((page - 1) * PageSize)
                .Take(PageSize)
            );
        }
    }
}